### PR TITLE
session: add index on 'user' field for mysql tables | tidb-test=pr/2430

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #55563 

Problem Summary:

### What changed and how does it work?

Another solution for the same purpose, see https://github.com/pingcap/tidb/pull/57320
If this one is chosen, that one could be closed.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)

create new cluster and upgrade old cluster, show create table to see the result.

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/session/bootstrap.go [hunk 1]
```diff
@@ -3272,6 +3280,20 @@ func upgradeToVer218(_ sessiontypes.Session, ver int64) {
...
+		return
+	}
+
+	mustExecute(s, "ALTER TABLE mysql.user ADD INDEX (user)")
+	mustExecute(s, "ALTER TABLE mysql.global_priv ADD INDEX (user)")
+	mustExecute(s, "ALTER TABLE mysql.db ADD INDEX (user)")
+	mustExecute(s, "ALTER TABLE mysql.tables_priv ADD INDEX (user)")
+	mustExecute(s, "ALTER TABLE mysql.columns_priv ADD INDEX (user)")
+	mustExecute(s, "ALTER TABLE mysql.global_grants ADD INDEX (user)")
+	mustExecute(s, "ALTER TABLE mysql.default_roles ADD INDEX (user)")
```
Comment (D3Hunter) [inline pkg/session/bootstrap.go:3294]: add index name and ignore `, dbterror.ErrDupKeyName` ?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Comment by tiancaiamao** [comment]
PTAL @bb7133

**Review by bb7133 — APPROVED** [review]
LGTM

File: pkg/session/bootstrap.go [hunk 2]
```diff
@@ -1232,13 +1237,16 @@ const (
 	version239 = 239
 
 	// version 240
-	// Add indexes to mysql.analyze_jobs to speed up the query.
+	// Add index on user field for some mysql tables.
 	version240 = 240
+
+	// Add indexes to mysql.analyze_jobs to speed up the query.
```
Comment (D3Hunter) [inline pkg/session/bootstrap.go:1243]: comments of 240/241 should be exchanged

**Review by D3Hunter — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**